app/src/main/java/com/kickstarter/services/KSApolloClientV2.kt

```diff
@@ -0,0 +1,148 @@
+package com.kickstarter.services
+
+import com.apollographql.apollo.ApolloCall
+import com.apollographql.apollo.ApolloClient
+import com.apollographql.apollo.api.Response
+import com.apollographql.apollo.exception.ApolloException
+import com.kickstarter.models.Project
+import com.kickstarter.models.StoredCard
+import com.kickstarter.services.mutations.SavePaymentMethodData
+import com.kickstarter.services.transformers.encodeRelayId
+import io.reactivex.Observable
+import io.reactivex.subjects.PublishSubject
+
+interface ApolloClientTypeV2 {
+    fun createSetupIntent(project: Project? = null): Observable<String>
+    fun savePaymentMethod(savePaymentMethodData: SavePaymentMethodData): Observable<StoredCard>
+    fun getStoredCards(): Observable<List<StoredCard>>
+    fun deletePaymentSource(paymentSourceId: String): Observable<DeletePaymentSourceMutation.Data>
+}
+
+class KSApolloClientV2(val service: ApolloClient) : ApolloClientTypeV2 {
+
+    override fun createSetupIntent(project: Project?): Observable<String> {
+        return Observable.defer {
+            val createSetupIntentMut = CreateSetupIntentMutation.builder()
+                .apply {
+                    if (project != null) this.projectId(encodeRelayId(project))
+                }
+                .build()
+
+            val ps = PublishSubject.create<String>()
+            this.service.mutate(createSetupIntentMut)
+                .enqueue(object : ApolloCall.Callback<CreateSetupIntentMutation.Data>() {
+                    override fun onFailure(exception: ApolloException) {
+                        ps.onError(exception)
+                    }
+
+                    override fun onResponse(response: Response<CreateSetupIntentMutation.Data>) {
+                        if (response.hasErrors()) ps.onError(java.lang.Exception(response.errors?.first()?.message))
+                        else {
+                            ps.onNext(response.data?.createSetupIntent()?.clientSecret() ?: "")
+                        }
+                        ps.onComplete()
```

Comment:
Notice that with RXJava 2, instead of  `ps.onCompleted()` we use `ps.onComplete()`